verification/tpm.go

```diff
@@ -0,0 +1,268 @@
+// Licensed under the Apache-2.0 license
+
+package verification
+
+import (
+	"crypto/ecdsa"
+	"crypto/sha256"
+	"crypto/sha512"
+	"crypto/x509"
+	"encoding/binary"
+	"flag"
+	"io"
+	"math"
+	"math/big"
+	"strings"
+	"testing"
+
+	"github.com/google/go-tpm-tools/client"
+	"github.com/google/go-tpm/legacy/tpm2"
+	"github.com/google/go-tpm/tpmutil"
+)
+
+var (
+	tpmPath                    = flag.String("tpm-path", "/dev/tpm0", "Path to the TPM device (character device or a Unix socket).")
+	tpmPolicySigningValidation = flag.String("tpm-policy-signing-validation", "disabled", "Validation of TPM policy signing with Caliptra DPE commands is enabled/disabled. By default, this is disabled.")
+
+	handleNames = map[string][]tpm2.HandleType{
+		"all":       {tpm2.HandleTypeLoadedSession, tpm2.HandleTypeSavedSession, tpm2.HandleTypeTransient},
+		"loaded":    {tpm2.HandleTypeLoadedSession},
+		"saved":     {tpm2.HandleTypeSavedSession},
+		"transient": {tpm2.HandleTypeTransient},
+	}
+)
+
+func TestTpmPolicySigning(d TestDPEInstance, c DPEClient, t *testing.T) {
+	if strings.ToLower(*tpmPolicySigningValidation) == "enabled" {
+		t.Log("Validation of TPM policy signing using Caliptra DPE commands is enabled.")
+		testTpmPolicySigning(d, c, t)
+	} else {
+		t.Log("Validation of TPM policy signing using Caliptra DPE commands is disabled, skipping this test.\n To enable this validation start the TPM emulator, set `go test . -tpm-policy-signing-validation=enabled` and rerun.")
+		return
+	}
+}
+
+func startTpmSession(t *testing.T, tpm io.ReadWriteCloser, alg tpm2.Algorithm) (tpmutil.Handle, []byte, error) {
+	totalHandles := 0
+	for _, handleType := range handleNames["all"] {
+		handles, err := client.Handles(tpm, handleType)
+		if err != nil {
+			t.Fatalf("[FATAL]: Error getting handles %s: %v", *tpmPath, err)
+		}
+		for _, handle := range handles {
+			if err = tpm2.FlushContext(tpm, handle); err != nil {
+				t.Fatalf("[FATAL]: Error flushing handle 0x%x: %v", handle, err)
+			}
+			t.Logf("Handle 0x%x flushed", handle)
+			totalHandles++
+		}
+	}
```

Comment:
nit: Would recommend splitting this into a helper `flushAllContexts`

Reply:
Added a separate method